src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts

```diff
@@ -1259,10 +1312,51 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, Component
     }
 
     /**
-     * Update adding existing questions from a file or course
+     * Update adding existing questions from a file or course or exam
      */
-    setExistingQuestionSourceToCourse(setToCourse: boolean): void {
-        this.showExistingQuestionsFromCourse = setToCourse;
+    setExistingQuestionSourceToCourse(): void {
+        this.showExistingQuestionsFromCourse = true;
+        this.showExistingQuestionsFromExam = false;
+        this.showExistingQuestionsFromFile = false;
+
+        this.selectedCourseId = undefined;
+        this.allExistingQuestions = this.existingQuestions = [];
+        this.importFile = undefined;
+        this.importFileName = '';
+        const control = document.getElementById('importFileInput') as HTMLInputElement;
+        if (control) {
+            control.value = '';
+        }
+        this.changeDetector.detectChanges();
```

Comment:
This is repeated in setExistingQuestionSourceToCourse, setExistingQuestionSourceToExam and setExistingQuestionSourceToFile.

Reply:
Thats a good points, removed the duplication in 5a9153e